### PR TITLE
Initial Life calendar interface and Canvas RRA

### PR DESCRIPTION
**Pull request by cdewitt02**
My initial life calendar work, idk if it still works with the new JSON strategy

### CODE REVIEW

**Review by ATRJR — DISMISSED** [review]
Looks good! Approved.

File: backend/calendar/user_events.py
Comment (Grayson34) [inline backend/calendar/user_events.py:1]: need to add more error checks in here. validate id's, event_id's and any other possible edge cases

Comment (Grayson34) [reply backend/calendar/user_events.py:1]: cant use absolute path to communicate with db, it wont work on our computers

File: backend/calendar/to_do_list.py
Comment (Grayson34) [inline backend/calendar/to_do_list.py:1]: absolute reference for db

File: backend/calendar/user_events.py
Comment (Grayson34) [reply backend/calendar/user_events.py:1]: needs function to fetch a persons events

File: backend/calendar/to_do_list.py
Comment (Grayson34) [reply backend/calendar/to_do_list.py:1]: some of these functions (like get_todo_list) isnt gonna be needed when we are just storing it all in db (it wll need to be written in the other file)

File: backend/calendar/canvasAPI/driver.py
Comment (ryanmercado) [inline backend/calendar/canvasAPI/driver.py:1]: we probably don't need this

File: backend/calendar/to_do_list.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+    def getList(self, usr_id):
+        con = sqlite3.connect('server/usrDatabase/usrDB.db')
+        cursor = con.cursor()
+        next_week_datetime = datetime.now() + timedelta(days=7)
+
+        query = """
+            SELECT event_id, event_title, start_epoch
+            FROM calendar
+            WHERE usr_id = ? AND start_epoch >= ? AND start_epoch <= ? AND is_important = 1
```
Comment (ryanmercado) [inline backend/calendar/to_do_list.py:29]: is_important -> on_to_do_list

File: backend/calendar/user_events.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        cursor = con.cursor()
+        cursor.execute('SELECT * FROM calendar WHERE usr_id = ?', (usr_id,))
+        events = cursor.fetchall()
+        con.close()
+        return events
+    else:
+        raise UserNotFoundException
+
+
+def update_event(event_id, new_event_title):
```
Comment (ryanmercado) [inline backend/calendar/user_events.py:48]: Change to update all required fields

**Review by ryanmercado — CHANGES_REQUESTED** [review]

**Review by ryanmercado — APPROVED** [review]
Looks good, Lets get grayson on still

File: backend/calendar/to_do_list.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
+from datetime import datetime, timedelta
+import user_events
+import sqlite3
+
+
+class ToDoList():
+
+    todo = [] # contains all the events to be shown in list
+
+    def get_todo_list(self, usr_id):
```
Comment (Grayson34) [inline backend/calendar/to_do_list.py:10]: is this needed

File: backend/calendar/to_do_list.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        self.todo = self.getList(usr_id)
+        return self.todo
+    
+    def add_to_list(self,usr_id, event_desc, event_type, event_title, start_epoch, end_epoch, on_to_do_list):
+        created_epoch = datetime.now().timestamp()
+        user_events.create_event(usr_id, 1, created_epoch, event_desc, event_type, event_title, start_epoch, end_epoch, on_to_do_list)
```
Comment (Grayson34) [inline backend/calendar/to_do_list.py:16]: event_id set to 1?

**Review by Grayson34 — APPROVED** [review]
barring the two tickets this can be approve

**Pull request merged**